File: 452-minimum-number-of-arrows-to-burst-balloons/solution.ts
function findMinArrowShots(points: number[][]): number {
  const sorted = points.sort((a, b) => a[0] - b[0]);
  let now: number[] = [];
  let answer = 0;

  sorted.forEach(([start, end]) => {
    if (now.length === 0) now = [start, end];
    else if (start <= now[1]) {
      now = [Math.max(start, now[0]), Math.min(end, now[1])];
    } else {
      answer++;
      now = [start, end];
    }
  });
  answer++;

  return answer;
}

File: 452-minimum-number-of-arrows-to-burst-balloons/solution.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        if(points.size() == 0) return 0;
        sort(points.begin(), points.end());
        int ans = 1;
        vector<int> curr = points[0];
        for(int i = 1; i < points.size(); i++) {
            if(points[i][0] <= curr[1]) {
                curr = {max(points[i][0], curr[0]), min(points[i][1], curr[1])};
            } else {
                ans++;
                curr = points[i];
            }
        }
        return ans;
    }
};
